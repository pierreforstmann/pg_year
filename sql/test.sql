--
-- test.sql
--
create extension pg_year;
--
create table t(c year);
--
insert into t values(1900);
insert into t values(2156);
insert into t values(1901);
insert into t values(2000);
insert into t values(2155);
delete from t where c = 1901 or c = 2155;
select c + 1 from t;
select c - 1 from t;
select from t  where c = 2000;
select from t  where c > 2000; 
select from t  where c >= 2000; 
select from t  where c < 2000;
select from t  where c <= 2000; 
select from t  where c <> 2000; 
--
create index b on t(c);
create index h on t using hash (c);
