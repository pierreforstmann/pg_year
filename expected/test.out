--
-- test.sql
--
create extension pg_year;
--
create table t(c year);
--
insert into t values(1900);
ERROR:  year cannot be lower than 1901
DETAIL:  value 1900 is lower than 1901
HINT:  make it greater or equal than 1901
insert into t values(2156);
ERROR:  year cannot be greater than 2155
DETAIL:  value 2156 is greater than 2155
HINT:  make it lower or equal than 2155
insert into t values(1901);
insert into t values(2000);
insert into t values(2155);
delete from t where c = 1901 or c = 2155;
select c + 1 from t;
 ?column? 
----------
 2001
(1 row)

select c - 1 from t;
 ?column? 
----------
 1999
(1 row)

select from t  where c = 2000;
--
(1 row)

select from t  where c > 2000; 
--
(0 rows)

select from t  where c >= 2000; 
--
(1 row)

select from t  where c < 2000;
--
(0 rows)

select from t  where c <= 2000; 
--
(1 row)

select from t  where c <> 2000; 
--
(0 rows)

--
create index b on t(c);
create index h on t using hash (c);
